generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/extensions/CompletableFutureExtensionsKtTest.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.generator.federation.extensions
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import reactor.kotlin.core.publisher.toMono
+import java.time.Duration
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+import java.util.concurrent.ExecutionException
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.test.assertTrue
+
+class CompletableFutureExtensionsKtTest {
+    @Test
+    fun `joinAll asynchronously collects a list of completable futures of elements into a completable future list of elements`() {
+        val firstPromise = "first promise".toMono().delayElement(Duration.ofMillis(500)).toFuture()
```

Comment:
`toMono()`? no love for coroutines? :P

Reply:
one of each :P, stream, coroutine and promise